internal/services/books/service.go

```diff
@@ -0,0 +1,317 @@
+package books
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"test-project/internal/models"
+	"test-project/internal/services/googlebooks"
+	"test-project/internal/util/db"
+
+	"github.com/rs/zerolog/log"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+)
+
+type Service struct {
+	db                 *sql.DB
+	googleBooksService *googlebooks.Service
+}
+
+func NewService(db *sql.DB, googleBooksService *googlebooks.Service) *Service {
+	return &Service{
+		db:                 db,
+		googleBooksService: googleBooksService,
+	}
+}
+
+func (s *Service) GetBookByID(ctx context.Context, bookID string) (*models.Book, error) {
+	logger := log.Ctx(ctx).With().Str("bookID", bookID).Logger()
+	book, err := models.FindBook(ctx, s.db, bookID)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			logger.Warn().Msg("book not found with ID: " + bookID)
+			return nil, fmt.Errorf("Book not found with ID: %s", bookID)
```

Comment:
du kannst daten dem logger per .Str() z.B. übergeben, dann kann man in den logs besser danach filtern